File: sorting/insertSort/insertSort.js
function insertSort(array){
    for (let index=0;index<array.length;index++){
        let value = array[index];
        let j = index-1;
        while (j>=0 && value < array[j]){
            array[j+1] = array[j];
            j--;
        };
        array[j+1] = value;
        console.log(array);
    };
    return array;

};

console.log(insertSort([1 ,4 ,234, 24 ,542, 52 ,76 ,234, 542 ,43]))

File: sorting/insertSort/insertSort.cpp
#include <iostream>
#include <vector>


using namespace std;

void insertSort(vector<int> &v){


for (int index{1};index<v.size();index++){
    int j{index-1};
    int value{v[index]};
    while (j>=0 && value < v[j]){
        v[j+1] = v[j];
        j--;
    };
    v[j+1] = value;
};


};


void display(const vector<int> &v){
    for (int i{};i<v.size();i++){
        cout << v[i] << " ";
    };
    cout << endl;

};

int main(){
    int size{};
    cout << "Enter the number of items in the vector : ";
    cin >> size;
    vector<int> mainVector(size);
    cout << "Enter the numbers separated by spaces : ";
    for (int i{};i<size;i++){
        
        cin >> mainVector[i];

    };
    display(mainVector);
    cout << "Sorting the array..." << endl;
    insertSort(mainVector);
    cout << "Sorted array -> " << endl;
    display(mainVector);
    return 0;
}

